llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -150,33 +170,109 @@ void InterBlockScheduling::leaveFunction() {
 }
 
 void InterBlockScheduling::enterBlock(MachineBasicBlock *BB) {
-  CurrentBlock = &getBlockState(BB);
-  CurrentBlock->resetRegion();
+  CurrentBlockState = &getBlockState(BB);
+  CurrentBlockState->resetRegion();
   DEBUG_BLOCKS(dbgs() << "  >> enterBlock " << BB->getNumber() << " "
-                      << CurrentBlock->kindAsString() << " FixPointIter="
-                      << CurrentBlock->FixPoint.NumIters << "\n");
+                      << CurrentBlockState->kindAsString() << " FixPointIter="
+                      << CurrentBlockState->FixPoint.NumIters
+                      << " II=" << CurrentBlockState->FixPoint.II);
 }
+namespace {
+class PipelineExtractor : public PipelineScheduleVisitor {
+  BlockState &Loop;
+  BlockState *Prologue = nullptr;
+  BlockState *Epilogue = nullptr;
+  MachineBundle CurrentBundle;
+  std::vector<MachineBundle> TimedRegion;
+  bool InLoop = false;
+
+  void startPrologue() override {
+    // Nothing at this time, but let's keep the override around
```

Comment:
Do we have some future plan for this method?

Reply:
Only vaguely. Currently we insert an opaque region which is not seen by the normal scheduler. In future we may need to do some boundary signaling.